src/test/java/org/killbill/billing/plugin/vertex/VertexTaxCalculatorTest.java

```diff
@@ -231,6 +231,95 @@ public void testTaxDescription() throws Exception {
         assertEquals("CA STATE TAX", result.get(0).getDescription());
     }
 
+    @Test(groups = "fast")
+    public void testTaxEffectiveRate() throws Exception {
+        //given
+        given(invoice.getInvoiceItems()).willReturn(Collections.singletonList(taxableInvoiceItem));
+        given(taxResponse.getData()).willReturn(apiResponseData);
+        final TaxesType taxesType = new TaxesType();
+        taxesType.setCalculatedTax(MOCK_TAX_AMOUNT_1_01);
+        given(responseLineItem.getTaxes()).willReturn(Collections.singletonList(taxesType));
+
+        //given tax effective rate is present
+        taxesType.setEffectiveRate(0.09975d);
+
+        //then it persisted in item details invoice item field
+        List<InvoiceItem> result = vertexTaxCalculator.compute(account, invoice, true, Collections.emptyList(), tenantContext);
+        assertEquals("{\"taxRate\":0.09975}", result.get(0).getItemDetails());
+        checkTaxItemFields(result.get(0));
+
+        //given effective rate is not present
+        taxesType.setEffectiveRate(null);
+
+        //then no item details persisted
+        result = vertexTaxCalculator.compute(account, invoice, true, Collections.emptyList(), tenantContext);
+        assertNull(result.get(0).getItemDetails());
```

Comment:
it is possible that taxRate is not 0 and amount is 0 but vice versa cannot be true.

Reply:
discussed